services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/utils/MongoDbDocumentBuilder.java

```diff
@@ -207,12 +207,9 @@ public MongoDbDocumentBuilder withDeviceFilters(final List<Filter> filters) {
         filters.forEach(filter -> {
             if (filter.getValue() instanceof String) {
                 final String value = (String) filter.getValue();
-                if (value.contains("*") || value.contains("?")) {
-                    // if the value contains * and ? then use regex matching
-                    document.put(mapDeviceField(filter.getField()),
-                            new JsonObject().put("$regex",
-                                    DeviceRegistryUtils.getRegexExpressionForSearchOperation(value)));
-                }
+                document.put(mapDeviceField(filter.getField()),
```

Comment:
would you mind creating an issue for this that we can refer to in this PR as this seems to be clearly a bug, right?

Reply:
I have created an issue  #2308 for the above.